### DISCUSSION TITLE
Customise list printing

### DISCUSSION THREAD

**Discussion by tekumara** [Textualize/rich#2898] [Q&A] [answered]
Is it possible to customise the behaviour of list printing?

Current behaviour:
```
>>> from rich.console import Console
>>> console = Console()
>>> console.print(["a", "b", "c"])
['a', 'b', 'c']
```
Instead I'd like to print the list with each element on a new line, eg:
```
a
b
c
```

**Comment by geekya215** [comment, marked as answer]
you should unpack the list parameter, and set the default separator to `\n`

```python
console.print(*["a", "b", "c"], sep='\n')
```

get more detailed information from [here](https://github.com/Textualize/rich/blob/076e0d208eb0b4e74cd8639e11a558b9319bd799/rich/console.py#L1644)

**Reply by tekumara** [reply]
Thank you, yes that would work... Is there a way to do this when printing a table cell? I simplified the problem a little bit when describing it above.

**Reply by geekya215** [reply]
Can you give me some examples?

**Reply by tekumara** [reply]
Say I have a list of fruits, eg: `fruit = ["apricot", "banana"]` ... I'd like to display them in a single cell each on its own line, eg:

```
      examples       
┏━━━━━━━━┳━━━━━━━━━━┓
┃ name   ┃ examples ┃
┡━━━━━━━━╇━━━━━━━━━━┩
│ fruits │ apricot  │
│        │ banana   │
└────────┴──────────┘
```

One way I guess, it is turn the list into a str first, eg:

```python
from rich.console import Console
from rich.table import Table

table = Table(title="example")
table.add_column("name")
table.add_column("examples")

fruit = ["apricot", "banana"]
table.add_row("fruits", "\n".join(fruit))

console = Console()
console.print(table)
```

But I wondered if Rich has a style or some other method for customising the rendering of lists in table cells.

**Reply by geekya215** [reply]
I think if you want to to display multiple similar elements in the same cell, a more appropriate way is to put them in multiple cells.

> But I wondered if Rich has a style or some other method for customising the rendering of lists in table cells.

The easiest way to render a list into multiple lines is to convert it to str, otherwise you need to implement the `__rich__` or `__rich_console__` methods for the list object. You can refer to the following information to customize the rendering logic of list.

- [Console Protocol](https://rich.readthedocs.io/en/stable/protocol.html)
- [`add_row` in rich/table.py](https://github.com/Textualize/rich/blob/076e0d208eb0b4e74cd8639e11a558b9319bd799/rich/table.py#L418-L463)
- [`print` in rich/console.py](https://github.com/Textualize/rich/blob/076e0d208eb0b4e74cd8639e11a558b9319bd799/rich/console.py#L1693-L1719)
